### ISSUE TITLE
Postgres regression: spans no longer include query

### ISSUE THREAD

**Issue by comp615** [DataDog/dd-trace-js#3007]
**Expected behaviour**
pg.query should include the query being run as part of the span information

**Actual behaviour**
pg.query no longer includes any information about the query; but the span itself (duration, etc.) is still logged.

Before:
<img width="687" alt="image" src="https://user-images.githubusercontent.com/542149/233680226-a8dd3360-5caa-418c-8260-81f4df9247d7.png">

After:
<img width="813" alt="image" src="https://user-images.githubusercontent.com/542149/233679783-40889e95-b9f7-465a-bbd5-a41d9b574b1b.png">

**Steps to reproduce**
Use `pg` package to query in conjunction with `dd-trace`

**Environment**
This change seemed to happen when we updated from `"dd-trace": "3.16.0"`  to `"dd-trace": "^3.17.0"`.

We do setup postgres via knex + objection.

**Comment by tlhunter** [comment]
It might be related to #2938 which was released in 3.17.0.

I don't suppose you have a before and after example to help nail down exactly what data is now missing?

**Comment by comp615** [comment]
I can kind of show before and after in Datadog. Previously it was the query itself, now it's just `pg.query`

Before:
<img width="687" alt="image" src="https://user-images.githubusercontent.com/542149/233680226-a8dd3360-5caa-418c-8260-81f4df9247d7.png">

After:
<img width="813" alt="image" src="https://user-images.githubusercontent.com/542149/233679783-40889e95-b9f7-465a-bbd5-a41d9b574b1b.png">

**Comment by pregnantboy** [comment]
downgrading to `dd-trace@3.16.0` works for me. also with knex+objection

**Comment by tlhunter** [comment]
Can you provide a really basic reproduction? I do wonder if this only happens when objection is involved. Our existing pg tests haven't shown any regressions but we don't have any tests that involve objection.

**Comment by mastermatt** [comment]
We are also experiencing this issue with an upgrade to 3.17.0, however, we are not using Objection. Just knex@0.21.17 directly.

**Comment by comp615** [comment]
To the start channel what was originally published (3.14) was a Query object with the properties text and values.
```
Query {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  text: 'select *',
    values: [
      '123',
      '456'
    ],
...
}
```

In 3.17 this changed to 
```
{
  text: {
    text: 'select *',
    values: [
      '123',
      '456'
    ]
  }
}
```

This you'll notice is a step above (it has a mistaken extra key). This appears to be somewhat of an interplay between knex and pg. Because at the very top of my call stack, I have a single correct query (likely where we use pg directly).

I think it's likely that the _arguments_ to queries vary if knex is in use or not, but by the time knex does its thing, the query is correct. Previously we seem to pull the text from the query, whereas now we use arguments which is not a valid approach

**Comment by comp615** [comment]
Fixes covered (I think) in #3087 and #3091
